exist-core/src/main/java/org/exist/storage/XQueryPool.java

```diff
@@ -182,7 +183,9 @@ public CompiledXQuery borrowCompiledXQuery(final DBBroker broker, final Source s
         }
 
         //check execution permission
-        source.validate(broker.getCurrentSubject(), Permission.EXECUTE);
+        if (source instanceof DBSource) {
+            ((DBSource) source).validate(Permission.EXECUTE);
+        }
```

Comment:
do we need some logging here, in  case the match fails?

Reply:
Same as above